Sources/Plasma/PubUtilLib/plProgressMgr/plProgressMgr.cpp

```diff
@@ -80,9 +81,13 @@ plProgressMgr::plProgressMgr()
     fCallbackProc = nullptr;
     fCurrentStaticText = kNone;
 
-    // Fill array with pre-computed loading frame IDs
-    for (int i=0; i < LOADING_RES_COUNT; i++)
-        fImageRotation[i] = ST::format("xLoading_Linking.{02}.png", i);
+    // Find linking-animation frame IDs and store the sorted list
+    std::regex re("xLoading_Linking\\.[0-9]+?\\.png");
+    for (auto name : plClientResMgr::Instance().getResourceNames()) {
+        if (std::regex_match(name.to_std_string(), re))
```

Comment:
You can avoid the conversion to `std::string` here by using `std::regex_match(name.begin(), name.end(), re)` instead

Reply:
Excellent, I didn't see that option when I was looking at the regex matches.  I was hoping to write something a bit better here.